packages/embed_api/src/createlyEmbed.ts

```diff
@@ -0,0 +1,187 @@
+        iframe.style.height = "100%";
+        iframe.style.position = "absolute";
+        iframe.src = `https://app.creately.com/diagram/${this.doc.docID}/${this.doc.docMode}/`;
+        iframe.id = this.doc.iFrameId;
+        document.body.appendChild(iframe);
+        this.documentLoad.pipe(
+        ).subscribe( value => {
+            if( value === true ) {
+                console.log('documentLoad', value);
+                this.postMessage.sendToWindow(event,message,this.doc.iFrameId);
+            }
+        });
+    }
+    /**
+     * subscribe to this function will give the result the base on event that user requested
+     */
+    public on( event:PostMessageSendEventType ) : Observable<unknown>{
+        if (Object.values(PostMessageSendEventType).includes(event)) {
+            const message =  {};
+            this.documentReady.subscribe( value => {
+                if( value === true ){
+                console.log('documentReady', value);
+                this.postMessage.sendToWindow(event,message,this.doc.iFrameId);}
+            });
+            return this.postMessage.recv().pipe(
+                switchMap( msg => this.handleIncomingMessages(msg))
+               );
+            }
+            return throwError('Event is not valid');
+        }
+
+    /**
+     * this function will able to modify the document user have to send the changeable properties to function
+     *     document.name = 'AAA';
+     * document.shapes.aaaaaaa.x = 100;
+     */
+    public modify( document:object ) {
+        const event = PostMessageSendEventType.documentModify;
+        const message =  {
+            modifier: { $set: document }
+        };
+        this.postMessage.sendToWindow(event,message,this.doc.iFrameId);
+       return this.postMessage.recv().pipe(
+         switchMap( msg => this.handleIncomingMessages(msg))
+        );
+    }
+    /**
+     * this can overwrite the exiting iframe src url
+     * @param baseUrl base url ex app.creatly.com
+     */
+    public setAppBaseUrl (baseUrl:string) {
+        let frame = document.getElementById(this.doc.iFrameId);
+        if (this.isIFrame(frame) && frame.contentWindow) {
+            frame.src = `https://${baseUrl}/diagram/${this.doc.docID}/${this.doc.docMode}/`;
+        }
+    }
+    /**
+     * this will check if given object is a iframe or not.base on that this will return a boolean.
+     * @param input Html Object
+     */
+    private isIFrame (input: HTMLElement | null):input is HTMLIFrameElement {
+        return input !== null && input.tagName === 'IFRAME';
+    }
+ /**
+  * this function will filter out the incoming postMessage base on the event.
+  * @param msg
+  */
+    public handleIncomingMessages( msg: any ) {
+        if ( !msg || typeof msg !== 'object' ) {
+            return of();
+        }
+        const { event, data } = msg as any;
+        if ( event === PostMessageRecvEventType.documentData) {
+            return of(data);
```

Comment:
When the event is PostMessageRecvEventType.documentData

the first emit is
```
{
    data: { ... },
    modifier: undefined,
}
```

all other emits are
```
{
    data: undefined,
    modifier: { $set: { ... } }
}
```

merge all other emits with the first emit so that the subscription emits a diagram model always